src/vaadin-context-menu.html

```diff
@@ -370,44 +384,58 @@
           this._boundOnGlobalContextMenu = this._onGlobalContextMenu.bind(this);
         }
 
+        /** @protected */
         connectedCallback() {
           super.connectedCallback();
 
           this.__boundOnScroll = this.__onScroll.bind(this);
           window.addEventListener('scroll', this.__boundOnScroll, true);
         }
 
+        /** @protected */
         disconnectedCallback() {
           super.disconnectedCallback();
 
           window.removeEventListener('scroll', this.__boundOnScroll, true);
           this.close();
         }
 
+        /** @protected */
         ready() {
           super.ready();
           this._observer = new Polymer.FlattenedNodesObserver(this, info => {
             this._setTemplateFromNodes(info.addedNodes);
           });
         }
 
+        /**
+         * @param {!Array<!Node>} nodes
+         * @protected
+         */
         _setTemplateFromNodes(nodes) {
```

Comment:
Why protected instead of private?

Reply:
This is similar to overlay https://github.com/vaadin/vaadin-overlay/blob/master/src/vaadin-overlay.html#L457-L463

But I'm not sure also. Maybe it could be private here?

Reply:
Yes it's protected to be aligned with the overlay. We will remove that in the next major release anyways, but maybe someone could rely on overriding this method now 🤔

Reply:
Just to clarify, I'm fine with having these protected.